### ISSUE TITLE
Accessibility: colour contrast check

### ISSUE THREAD

**Issue by saschaleib** [saschaleib/dokuwiki-template-ad-hominem#52]
It occurred to me that I never made a systematic check of all combinations of foreground and background colours if they meet the [WCAG 2.2 level AA](https://www.w3.org/WAI/WCAG22/quickref/?versions=2.2&currentsidebar=%23col_customize&levels=aaa#top) requirements for a colour contrast of at least 4.5:1 (see: [1.4.3 Contrast (Minimum)](https://www.w3.org/WAI/WCAG22/quickref/?versions=2.1&showtechniques=131#contrast-minimum)).

So I used [WebAIM Contrast Checker](https://webaim.org/resources/contrastchecker/) to map out all possible combinations:

### Light mode colours:

| Variables |  | background | background_alt | background_neu | background_site | highlight |
| --- | --- | --- | --- | --- | --- | --- |
|  | Colours | `#FFFFFF` | `#EEEEEE` | `#DDDDDD` | `#fbfaf9` | `#fcffaa` |
| text | `#111111` | 18.88:1 | 16.27:1 | 13.9:1 | 18.11:1 | 18.00:1 |
| text_alt | `#333333` | 12.63:1 | 10.88:1 | 9.30:1 | 12.11:1 | 12.04:1 |
| text_neu | `#666666` | 5.74:1 | 4.94:1 | **4.22:1** (_fail_) | 5.50:1 | 5.47:1 |
| headlines | `#333333` | 12.63:1 | 10.88:1 | 9.30:1 | 12.11:1 | 12.04:1 |
| link | `#1F53A8` | 7.34:1 | 6.32:1 | 5.40:1 | 7.04:1 | 6.99:1 |
| existing | `#006500` | 7.33:1 | 6.32:1 | 5.40:1 | 7.03:1 | 6.99:1 |
| missing | `#A92700` | 7.05:1 | 6.07:1 | 5.19:1 | 6.76:1 | 6.72:1 |
| inpage | `#494F52` | 8.32:1 | 7.17:1 | 6.12:1 | 7.98:1 | 7.93:1 |

Only the combination of _neutral background_ with _neutral text_ colour misses the mark of at least 4.5:1. Both colours are not much used and only included for compatibility. Still the colour should be corrected.

### Dark mode colours:

There are no _neutral_ colours defined for dark mode, and the `highlight_dark` colour is only used in the context of code blocks with syntax highlighting. The full test looks as follows:

| Variables |  | bg_dark | bg_alt_dark | bg_site_dark | highlight_dark |
| -- | --- | --- | --- | --- | --- |
|  | Colours | `#232327` | `#39393D` | `#0C0C0D` | `#3c3c36` |
| text_dark | `#CCCCCC` | 9.75:1 | 7.15:1 | 12.17:1 | 6.91:1 |
| text_alt_dark | `#AEAEAE` | 7.05:1 | 5.18:1 | 8.81:1 | 5.00:1 |
| link_dark | `#76B0FD` | 7.00:1 | 5.14:1 | 8.75:1 | 4.96:1 |
| existing_dark | `#00CA1B` | 7.05:1 | 5.18:1 | 8.81:1 | 5.00:1 |
| missing_dark | `#FF9233` | 7.01:1 | 5.15:1 | 8.76:1 | 4.97:1 |
| inpage_dark | `#A7AEA6` | 6.89:1 | 5.06:1 | 8.61:1 | 4.88:1 |

All dark mode colour combinations have at least 4.5:1 colour contrast.

### Changes

The only change required is the _neutral_ text colour for _light_ mode. A value of `#616161` should solve this issue, as shown in the test results below:

| Variables |  | background | background_alt | background_neu | background_site | highlight |
| --- | --- | --- | --- | --- | --- | --- |
|  | Colours | `#FFFFFF` | `#EEEEEE` | `#DDDDDD` | `#fbfaf9` | `#fcffaa` |
| text_neu (changed) | `#616161` | 6.19:1 | 5.33:1 | 4.55:1 | 5.94:1 | 5.90:1 |

**Comment by saschaleib** [comment]
Rolled out today.
